src/read_data.cpp

```diff
@@ -1856,8 +1856,12 @@ void ReadData::open(char *file)
   if (!compressed) fp = fopen(file,"r");
   else {
 #ifdef LAMMPS_GZIP
-    char gunzip[128];
-    sprintf(gunzip,"gzip -c -d %s",file);
+    char gunzip[2048];
+    // Use taskset to force the gzip process to NOT run on the 0th "CPU", which should
+    // keep it from thrashing with the MPI rank zero process (the one reading the pipe).
+    // This is Linux specific, and the 1023 upper range might also be system specific.
+    // Use of something like hwloc would be more portable... but more complicated.
+    sprintf(gunzip,"taskset -c 1-1023 gzip -c -d %s",file);
```

Comment:
This is such a simple hack... and very effective.  I wonder if I could get Axel to look into a more portable way to do this, by simply making the pull request. ;-)